internal/infrastructure/ceph/rbd.go

```diff
@@ -149,26 +194,79 @@ func applyDiffToRawImage(
 
 	// Open raw image file
 	var rawImgFile *os.File
+	var rawChecksumFile *os.File
+
 	if _, err := os.Stat(rawImageFilePath); os.IsNotExist(err) {
-		rawImgFile, err = os.OpenFile(rawImageFilePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+		// Note: The checksum file must be updated first. If it is not updated before writing raw.img,
+		// and a crash occurs while updating the checksum, the next retry will fail checksum verification.
+		// By updating the checksum first, we ensure that if a crash occurs, the file can be safely
+		// recreated on the next attempt.
+		if enableChecksumVerify {
+			checksumFilePath := getChecksumFilePath(rawImageFilePath)
+			rawChecksumFile, err = os.OpenFile(checksumFilePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+			if err != nil {
+				return fmt.Errorf("failed to create checksum file: %s: %w", checksumFilePath, err)
+			}
+			defer func() { _ = rawChecksumFile.Close() }()
+
+			numChunks := expansionUnitSize / rawChecksumChunkSize
+			if expansionUnitSize%rawChecksumChunkSize != 0 {
+				numChunks++
+			}
+			for i := uint64(0); i < numChunks; i++ {
+				if err := writeChecksumAtChunk(rawChecksumFile, i, csumZero); err != nil {
+					return fmt.Errorf("failed to initialize checksum file: %w", err)
+				}
+			}
```

Comment:
It seems the checksum may be written halfway, that is, there may be a case when the checksum for chunk #0 through #10 are written but not for #11 to #20.